Add guidance on alignment of wizard elements (#2973)
grommet/hpe-design-system: aries-site/src/pages/templates/wizard.mdx

  <WizardValidationExample />
</Example>

## When to use

Use a wizard when your form involves multiple steps. This makes the experience more manageable for the user and helps to eliminate error by validating inputs as the user progresses.

## Anatomy

<Example plain>
  <WizardAnatomy />
</Example>

> 1. **Wizard header**: This region is intended to provide user with context on what the wizard is about.

Reviewer: ```suggestion
1. **Wizard header**: This region identifies the task flow and provides the wizard's controls.
```

Reviewer: Should we also mention the header spans the full width of the viewport?

>    - **a. Wizard title** (required): A short and descriptive title that identifies the wizard.

Reviewer: Active voice.

```suggestion
   - **a. Wizard title** (required): A short, descriptive title identifying the wizard.
```

Author: Incorporated in latest commit.

>    - **b. Previous step button** (required): A button that allows the user to move back one step in the wizard. This button is aligned to the left side of the header and is present on all steps expect the first.

Reviewer: ```suggestion
   - **b. Previous step button** (required): A button allowing the user to return to the previous wizard step. This button is positioned similarly to a browser's back button and is aligned to the left side of the wizard's header. It should appear on all steps except the first.
```

Author: Incorporated in latest commit.

>    - **c. Cancel button** (required): A button that allows the user to leave the wizard. When clicked, a confirmation modal will appear. This button is aligned to the right side of the header.

Reviewer: ```suggestion
   - **c. Cancel button** (required): A button allowing the user to leave the wizard. When clicked, a confirmation modal will appear. This button is aligned to the right side of the header.
```

Author: Incorporated in latest commit.

> 1. **Step identifier:** This region is intended to provide the user with context on what the step is about.

Reviewer: ```suggestion
1. **Step identifier:** This region provides the user with information regarding the current step.
```

Author: Incorporated in latest commit.

>    - **a. Step title** (required): A short and descriptive title that uniquely identifies the step.

Reviewer: ```suggestion
   - **a. Step title** (required): A short, descriptive title uniquely identifying the step.
```

Author: Incorporated in latest commit.

   - **b. Step summary** (optional): A summary of where the user is within the wizard, formatted as "Step X of Y".

>    - **c. Step description** (optional): Any additional context or information about the given step.

Reviewer: ```suggestion
   - **c. Step description** (optional): Any additional context or information about the current step.
```

Author: Incorporated in latest commit.

1. **Step inputs**: Form inputs for the current step.

> 1. **Step footer**: The footer contains a primary action button to advance to the next step or finish the wizard. The footer should remain a consistent width across all steps of a wizard. Its width should match the width of the widest step.

Reviewer: ```suggestion
1. **Step footer**: The footer contains a primary action button to advance to the next step or finish the wizard. The footer should remain a consistent width across all steps of a wizard so that the next button is placed consistently for the user. Its width should match the width of the widest step.
```

Author: Incorporated in latest commit.

   - **a. Next/finish button** (required): A primary action button that allows the user to advance to the next step or finish the wizard. This button should be aligned to the far-right side of the footer.

> 1. **Cancel wizard modal** (required): A center modal that provides double confirmation before the wizard is exited and cancelled. It appears when the user clicks on the "Cancel" button.

Reviewer: Should we mention something about "not displayed?"

Should we eventually add an anatomy diagram for the "cancel wizard modal?"

Author: I'll mention "Not displayed" for now and we can handle the added anatomy diagram separately.

Author: Incorporated in latest commit.

- On desktop, place the guidance to the right of form inputs.
- On mobile, place the guidance above form inputs.

> ## Alignment

Reviewer: Would the anatomy treatment be a more digestible and consistent way for presenting this type of information?

Author: [Yes, agree](https://github.com/grommet/hpe-design-system/pull/2973#issuecomment-1281324168). I was trying to have this be a quick win last week by starting with some written guidance but since it carried over to this week I can add an anatomy diagram into this page.

Author: Done.

> 

Reviewer: There has also been confusion to where the "next button" should be aligned when content between steps is different widths. This came up in a recent office hours.

I believe the first example says "keep the wizard width consistent between steps so that the next button placement does not change between each step" or something along those lines. Can we address that guidance here to make it more clear?

Author: Include this in the step footer description of anatomy diagram.

### Previous step button

A button that enables the user to navigate to the previous step should be placed in the wizard header and aligned to the far-left side of the header.

### Cancel button

> A button that enables the user to cancel out of the wizard should be placed in the wizard header and aligned to the far-right side of the header.

Reviewer 2: is Wizard capital?

Author: I'm never sure when to capitalize or not, but I think because this is just referring more generically to the concept of a wizard (not a formalized component) then it's okay to be lower case.

Reviewer 2: okay fair also not exactly sure.

### Title

The title should be centered in the wizard header.

### Next step button

> A primary action button that enables the user to advance to the next step should be placed in the footer of the wizard step and aligned to the far-right side of the footer.

Reviewer 2: a primary CTA button?

Author: CTA button refers to the one with HPE element, in this case it's just primary action.

Reviewer 2: gotcha thanks
